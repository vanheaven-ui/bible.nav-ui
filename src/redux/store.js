import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunkMiddleware from 'redux-thunk';
import rootReducer from './reducers';

const middlewareEnhancer = applyMiddleware(thunkMiddleware);
const composedEnhancer = composeWithDevTools(middlewareEnhancer);

const store = createStore(rootReducer, undefined, composedEnhancer);

export default store;
