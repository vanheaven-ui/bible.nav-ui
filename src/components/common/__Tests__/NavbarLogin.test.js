import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import rootDiv from '../../../testHelper';
import store from '../../../redux/store';
import '@testing-library/jest-dom';
import LoggedInNavbar from '../NavbarLogin';

it('renders without crashing', () => {
  const status = () => true;

  ReactDOM.render(
    <Provider store={store}><Router><LoggedInNavbar status={status} /></Router></Provider>, rootDiv(),
  );
});
