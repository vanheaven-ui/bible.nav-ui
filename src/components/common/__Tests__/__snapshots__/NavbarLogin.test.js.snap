// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarLoggedIn Component does not change unexpectedly 1`] = `
<nav
  className="navbar"
>
  <div
    className="nav-container"
  >
    <a
      className="nav-logo"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        height="50px"
        src="logo1.png"
        width="50px"
      />
      Bible.nav
    </a>
    <ul
      className="nav-menu"
    >
      <li
        className="nav-item profile"
      >
        <span>
          <small>
            ~/
          </small>
           
        </span>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-links"
          href="/favorites"
          onClick={[Function]}
        >
          Favourites
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-links"
          href="/"
          onClick={[Function]}
        >
          Logout
        </a>
      </li>
    </ul>
    <button
      className="nav-icon"
      onClick={[Function]}
      type="button"
    >
      <i
        aria-label="times"
        className="fas fa-bars"
      />
    </button>
  </div>
</nav>
`;
